orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                                       "%s:%d Not able to add provider conduit_id ",__FILE__,__LINE__);
+                   return ORTE_ERROR;
+        }
+        opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                     "\n provider conduit_id : %d",orte_rml_ofi.ofi_conduits[prov_num].conduit_id);
+        opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                            "\n loading the attribute ORTE_PROV_NAME");
+        if( ORTE_SUCCESS == 
+            (ret = orte_set_attribute( ofi_prov, ORTE_PROV_NAME, ORTE_ATTR_GLOBAL,cur_fi->fabric_attr->prov_name ,OPAL_STRING))) {
+            opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                                "\n loading the attribute ORTE_PROTOCOL %s",fi_tostr(&cur_fi->ep_attr->protocol,FI_TYPE_PROTOCOL));
+            if( ORTE_SUCCESS == 
+                (ret = orte_set_attribute( ofi_prov, ORTE_PROTOCOL, ORTE_ATTR_GLOBAL,(void *)&cur_fi->ep_attr->protocol ,OPAL_UINT32))) {
+               // insert the opal_list_t into opal_value_t list
+               opal_value_load(providers_list,ofi_prov,OPAL_PTR);
+               opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                                  "\n loading the provider opal_list_t* prov=%x into opal_value_t list successful",
+                                  ofi_prov);
+            } else {
+                   opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                                       "%s:%d Not able to add provider name ",__FILE__,__LINE__);
+                   return ORTE_ERROR;
+            }
+        } else {
+               opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                                   "%s:%d Not able to add provider name ",__FILE__,__LINE__);
+               return ORTE_ERROR;
+        }
+
+        prev_provider = providers_list;
+        cur_fi = cur_fi->next;
+    }
+
+    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                        "\n%s:%d Completed Query Interface",__FILE__,__LINE__);
+    return ORTE_SUCCESS;
+}
+
+
+/*debug routine to print the opal_value_t returned by query interface */
+void print_transports_query()
+{
+    opal_value_t *providers=NULL;
+    char* prov_name = NULL;
+    int ret;
+    int32_t *protocol_ptr, protocol;
+    int8_t* prov_num;
+
+    protocol_ptr = &protocol;
+
+    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                        "\n print_transports_query() Begin- %s:%d",__FILE__,__LINE__);
+    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                        "\n calling the orte_rml_ofi_query_transports() ");
+    if( ORTE_SUCCESS == orte_rml_ofi_query_transports(&providers))
+    {
+        /*opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                "\n query_transports() completed, printing details\n"); */
```

Comment:
For a first commit of new functionality like this, we probably don't need to include commented-out code like this (similar for the commented-out code below).